### PR TITLE
Feature/write tests

### PR DESCRIPTION
**Pull request by blessedjasonmwanza**
## Feature/write tests
In this Test Driven Development (TDD) project, @afizsavage  and I (@blessedjasonmwanza ) Collaborated to:

- Write tests to check if the Solver instance is initiated successfully as an object
- Test ```Factorial method``` as  a Unit
- Implement and write test cases for the famous ```FizzBuzz``` challenge in Ruby

### CODE REVIEW

File: spec/solver_spec.rb [hunk 1]
```diff
@@ -0,0 +1,65 @@
...
+      expect(solver.fizzbuzz(3)).to eq 'fizz'
+    end
+
+    it 'returns buzz when n is divisible by 5' do
+      expect(solver.fizzbuzz(5)).to eq 'buzz'
+    end
+
+    it 'returns fizzbuzz when n is divisible by 3 and 5' do
+      expect(solver.fizzbuzz(15)).to eq 'fizzbuzz'
+    end
```
Comment (MrBrN197) [inline spec/solver_spec.rb:63]: - [ ] three of your test are currently not passing. please ensure that all your test are passing and proceed to submit for another review.

**Review by MrBrN197 — CHANGES_REQUESTED** [review]
## Hi, @blessedjasonmwanza  👋 

You've done a really nice job so far.
There are some issues that you still need to work on to go to the next project but you are almost there!
## To highlight 👏👏
- ✔ Linters checks passing correctly
- ✔factorial, fizzbuz & reverse methods implemented correctly.

## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: spec/solver_spec.rb (hunk 1 above)
Comment (blessedjasonmwanza) [reply spec/solver_spec.rb:63]: Hello @MrBrN197, Thanks a million for such a professional review. On to making the changes and requesting for another review.

**Review by bimbolabuari — APPROVED** [review]
Hi Team 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 



Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**